primefaces/src/main/java/org/primefaces/component/tabview/TabViewRenderer.java

```diff
@@ -253,26 +253,33 @@ protected boolean encodeTabHeader(FacesContext context, TabView tabView, Tab tab
         //header container
         writer.startElement("li", tab);
         writer.writeAttribute("class", styleClass, null);
-        writer.writeAttribute("role", "tab", null);
-        writer.writeAttribute(HTML.ARIA_EXPANDED, String.valueOf(active), null);
-        writer.writeAttribute(HTML.ARIA_SELECTED, String.valueOf(active), null);
-        writer.writeAttribute(HTML.ARIA_LABEL, tab.getAriaLabel(), null);
-        writer.writeAttribute("data-index", index, null);
+        writer.writeAttribute("role", "presentation", null);
         if (tab.getTitleStyle() != null) {
             writer.writeAttribute("style", tab.getTitleStyle(), null);
         }
         if (tab.getTitletip() != null) {
             writer.writeAttribute("title", tab.getTitletip(), null);
         }
-        if (tabindex != null) {
-            writer.writeAttribute("tabindex", tabindex, null);
-        }
 
         //title
+        String tabHeaderId = tab.getClientId(context) + "_header";
+        String tabContentId = tab.getClientId(context) + "_content";
```

Comment:
@melloware @tandraschko handled it this way, thoughts?

Reply:
Can you extract `tabGetClientId` to the line above so its only asked for once ` String clientId = tab.getClientId(context);`  So like this...

```java
String clientId = tab.getClientId(context);
String tabHeaderId = clientId + "_header";
String tabContentId = clientId  + "_content";
```